### PR TITLE
NOJIRA add the ability to only import a subset of records from a MySQL T...

### PR DESCRIPTION
**Pull request by kehh**
...able

### CODE REVIEW

**Comment by kehh** [comment]
Don't merge this - is_int doesn't cast to integer ;-0

File: app/lib/ca/Import/DataReaders/MySQLDataReader.php [hunk 1]
```diff
@@ -88,8 +88,15 @@ public function read($ps_source, $pa_options=null) {
 			if (!$this->ops_table) { 
 				return false;
 			}
-			
-			$this->opo_rows = $this->opo_handle->query("SELECT * FROM {$this->ops_table}");
+			$vs_limit = '';
+			if(isset($va_path['limit']) && is_integer($va_path['limit'])){
```
Comment (leftclickben) [inline app/lib/ca/Import/DataReaders/MySQLDataReader.php]: Try is_numeric()

is_integer() is a runtime type check, its return value is based on the actual type of the parameter, not its contents.

**Comment by kehh** [comment]
used caGetOption instead

File: app/lib/ca/Import/DataReaders/MySQLDataReader.php [hunk 2]
```diff
@@ -88,8 +88,17 @@ public function read($ps_source, $pa_options=null) {
...
 			}
-			
-			$this->opo_rows = $this->opo_handle->query("SELECT * FROM {$this->ops_table}");
+			$vs_limit = '';
+			$vn_limit = caGetOption('limit', $va_path, 0, array('castTo' => 'int'));
+			$vn_offset = caGetOption('offset', $va_path, 0, array('castTo' => 'int'));
+			if($vn_limit){
+				$vs_limit =  "$vn_limit";
+				if($vn_offset){
+					$vs_limit = "$vn_limit, $vs_limit";
```
Comment (leftclickben) [inline app/lib/ca/Import/DataReaders/MySQLDataReader.php]: This should be `$vs_limit = "$vn_offset, $vn_limit";` but also, this whole section (line 91 plus lines 94-100) could be rewritten as:

```
$vs_limit = $vn_limit ? (" LIMIT " . ($vn_offset ? "$vn_offset, $vn_limit" : $vn_limit)) : "";
```

(EDIT: Originally had offset and limit the wrong way around in the SQL)

File: app/lib/ca/Import/DataReaders/MySQLDataReader.php [hunk 3]
```diff
@@ -69,7 +69,7 @@ public function __construct($ps_source=null, $pa_options=null){
 	 * @return bool
 	 */
 	public function read($ps_source, $pa_options=null) {
-		# mysql://username:password@localhost/database?table=tablename
+		# mysql://username:password@localhost/database?table=tablename&limit=100&offset=10
```
Comment (leftclickben) [inline app/lib/ca/Import/DataReaders/MySQLDataReader.php:74]: Should limit and offset be somehow marked as optional?

**Pull request merged**